test/api/test_config.py

```diff
@@ -0,0 +1,277 @@
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or (at
+your option) any later version.
+
+This program is distributed in the hope that it will be useful, but
+WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program. If not, see https://www.gnu.org/licenses/
+"""
+
+import os
+import unittest
+from tempfile import TemporaryDirectory
+
+from core import typing as T, time
+from core.config import exception
+from api.config import Config, _validate, _schema
+
+_not_a_config = """ This is an example """
+_bad_YAML = """
+identity:
+    ldap:
+            host: baz
+        port: quux
+"""
+_basic_config = """
+identity:
+    ldap:
+        host: ldap.example.com
+    users:
+        dn: ou=users,dc=example,dc=com
+        attributes:
+            uid: uidNumber
+    groups:
+        dn: ou=groups,dc=example,dc=com
+        attributes:
+            gid: gidNumber
+persistence:
+    postgres:
+        host: postgres.example.com
+    database: sandman
+    user: a_db_user
+    password: abc123
+email:
+    smtp:
+        host: mail.example.com
+    sender: vault@example.com
+deletion:
+    threshold: 90
+archive:
+    threshold: 1000
+    handler: /path/to/executable
+"""
+_added_info_config = """
+identity:
+    ldap:
+        host: ldap.example.com
+    users:
+        dn: ou=users,dc=example,dc=com
+        attributes:
+            uid: uidNumber
+    groups:
+        dn: ou=groups,dc=example,dc=com
+        attributes:
+            gid: gidNumber
+persistence:
+    postgres:
+        host: postgres.example.com
+    database: sandman
+    data: information
+    user: a_db_user
+    password: abc123
+email:
+    smtp:
+        host: mail.example.com
+    sender: vault@example.com
+deletion:
+    threshold: 90
+    warnings:
+    - 240
+    - 72
+    - 24
+archive:
+    threshold: 1000
+    data: more information
+    handler: /path/to/executable
+"""
+_scalar_list_config = """
+identity:
+    ldap:
+        host: ldap.example.com
+    users:
+        dn: ou=users,dc=example,dc=com
+        attributes:
+            uid: uidNumber
+    groups:
+        dn: ou=groups,dc=example,dc=com
+        attributes:
+            gid: gidNumber
+persistence:
+    postgres:
+        host: postgres.example.com
+        port:
+            - 1
+            - 2
+    database: sandman
+    data: information
+    user: a_db_user
+    password: abc123
+email:
+    smtp:
+        host: mail.example.com
+    sender: vault@example.com
+deletion:
+    threshold: 90
+    warnings:
+    - 240
+    - 72
+    - 24
+archive:
+    threshold: 1000
+    data: more information
+    handler: /path/to/executable
+"""
+_incorrect_type_config = """
```

Comment:
I might be missing something, but I don't see any incorrect type in this data. The only thing that looks "odd" is that you've set `archive.handler` to 1234. However, that's still a valid path. However, it passes your test by virtue of the YAML parser converting `1234` into an integer, which isn't a valid input for `pathlib.Path` (it should be a string)... i.e., More luck than judgement.

Keep it simple and set something that should be an integer (e.g., a port number) to some random string.